packages/next-server/server/next-server.ts

```diff
@@ -266,8 +281,27 @@ export default class Server {
       return
     }
 
-    const resolver = interopDefault(require(resolverFunction))
-    resolver(req, res)
+    try {
+      // Parsing of cookies
+      req.cookies = parseCookies(req.headers.cookie || '')
+      // Parsing query string
+      req.query = parseQuery(req)
+      // // Parsing of body
+      req.body = await parseBody(req)
```

Comment:
Couldn't we add a get modifier to `req` and when `body` is accessed then we parse the body instead of parsing every time even if it's not used 🤔We could also add a `bodyOriginal` or something to allow custom handling of `body`

Reply:
Since the body hasn't been fully received yet when the handler is called, it has to be asynchronous. Or do you suggest it to be a getter for a promise?
```js
const body = await req.body;
```

Reply:
I agree we should ship it in the first version. There is no need to opt out from this approach, we could, later on, add config for different parsers.   But if we go by this approach we are going to move a little bit from `express`, because we need to resolve promise for `body` in the handler.

Reply:
Yeah, I think being a little different from express on this is ok since it will be more performant. We will just need to make sure the differences are clear when we document it

Reply:
I don't think we should add an alternative API. Keep in mind that if people were using bodyparser they would run it for every single request too, even when not used.